app/_hub/kong-inc/jq/index.md

```diff
@@ -0,0 +1,94 @@
+---
+name: Kong jq
+publisher: Kong Inc.
+versions: 0.1.x
+beta: true
+
+desc: Transform JSON objects included in API requests or responses using jq filters and programs.
+description: |
+    The Kong jq plugin enables arbitrary jq transformations on JSON objects included in API requests or responses. The source of the object can either be the request or response body, and the transformed result can either replace the body or be used to set HTTP headers.
+
+    Multiple filters can be specified in the configuration. You can use these filters to set request headers from values present in the request body, and then rewrite the response body with a separate jq program.
```

Comment:
This is no longer true. The transformed result now only affects the body - we had to drop the HTTP headers target as part of the simplification for technical reasons. Previously you could also add an arbitrary number of filters with different sources / targets, including headers. Now there are just two filters or "programs"; a request program which transforms the request body, and a response program which transforms the response body.

Here's what I wrote in the README for the simplified version, not sure if it's too much for a description field or not but it covers everything it can do. The last paragraph about buffering is perhaps important in complex setups, but most people probably won't need to worry about it.

```suggestion
    The Kong jq plugin enables arbitrary jq transformations on JSON objects included in API requests or responses.
    
    The configuration accepts two sets of options; one for the request and another for the response. In both cases a jq program string can be given, along with some jq option flags and a list of media types, one of which must be found in the `Content-Type` header in order for the jq program to run (defaults to `application/json`).

    In the response context, there is also the option to specify a list of status codes, one of which must match the response status code (defaults to `200`).

    Note that in the response context the entire body must be buffered in order to be processed. This also implies that the `Content-Length` header will be dropped if present, and the body transferred with chunked encoding.
```

Reply:
will update